File: movietracker.api/movietracker.api/src/main/java/com/mme/movietracker/api/repository/WatchlistRepository.java
package com.mme.movietracker.api.repository;

import com.mme.movietracker.api.domain.Watchlist;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface WatchlistRepository extends JpaRepository<Watchlist, Long> {
    List<Watchlist> findByNameContaining(String name);
}

File: MovieTracker.API/MovieTracker.Infra/Respositories/WatchlistRepository.cs
using MovieTracker.Core.Domain;
using MovieTracker.Core.Repositories;
using System.Data.Entity;
using System.Linq;

namespace MovieTracker.Infra.Respositories
{
    public class WatchlistRepository : Repository<Watchlist>, IWatchlistRepository
    {
        public WatchlistRepository(MovieTrackerContext context) : base(context) { }

        public Watchlist GetWatchlistWithWatchlistItems(long id)
        {
            return MovieTrackerContext.Watchlists
                .Include(x => x.WatchlistItems)
                .Where(x => x.Id == id)
                .First();
        }

        public MovieTrackerContext MovieTrackerContext
        {
            get { return Context as MovieTrackerContext; }
        }
    }
}